### PR TITLE
Create list of card previews on the edit/create flashcard tab

### LINKED ISSUE #21: ECF.1 Add list of Card Previews
## User Story 
As a user, I want the front and back of cards to be previewed in a list so I can see all of them in a set.

## Sub-tasks 
- [x] Card preview component 
  - If you want to use the html and scss of the flashcard component, we may need to have a base flashcard component and also a flippable-flashcard-component. Or idk somehow stop the flipping.
  - Front/back and hamburger (no functionality needed)
- [x] Scrollable list of previews
- [x] Selected card indicators

## Notes
- depends on #20 
Image
- On Edit Create Study Set page, in Flasacard-editor component
- You do not need to implement adding cards

### PR DESCRIPTION
**Pull request by ashkjones**
Fixes #21 and #23
On initiation, edit/create page gets a StudySet associated with an Id passed as an input. If no Id is passed, it creates a blank study set with a single flashcard in it. If you want to test what it looks like with a non-empty set, do the following:

1. edit line 28 in `edit-create-study-set.component.ts` to:
```
@Input() setId: string | undefined = "aaaa";
```
2. run in the terminal `json-server dev-set-db.json`

Selecting a card will make it's contents appear in the editor (the edit functionality will be added in #24 )

Dragging is implemented but is a little wonky if the screen size is too small due to the text overflowing. I am going to say it isn't worth trying to fix due our time constraints.

### CODE REVIEW

File: src/app/ec-card-preview/ec-card-preview.component.scss [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  text-align: center;
+}
+
+.preview-row{
+  display: flex;
+  flex: 1;
+  gap: 8px;
+}
+
+mat-icon {
```
Comment (B-Yamawaki) [inline src/app/ec-card-preview/ec-card-preview.component.scss:14]: This causes the flip symbol on the other flashcards to change their position. Could you try to target only the menu icon in this component.

**Review by B-Yamawaki — CHANGES_REQUESTED** [review]
I am very impressed with the card selection, moving, and editing. It is very cool. There are a few issues that mostly have to do with cleaner code.

File: src/app/ec-card-preview/ec-card-preview.component.scss (hunk 1 above)
Comment (ashkjones) [reply src/app/ec-card-preview/ec-card-preview.component.scss:11]: I realize the entire component style was being imported into styles.scss. I moved the mixins into their own file so only the mixins get imported into styles.scss. I followed steps 1-4 from an [Angular Material Tutorial](https://material.angular.io/guide/theming-your-components).

I used the naming scheme (i.e., _componentname-theme.scss) used by google angular github.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -5,13 +5,19 @@ import { MatSlideToggleModule } from '@angular/material/slide-toggle';
 import { FormsModule } from '@angular/forms';
 import { NavbarComponent } from './navbar/navbar.component';
 import { SignInComponent } from './sign-in/sign-in.component';
+import { EditCreateStudySetComponent } from './edit-create-study-set/edit-create-study-set.component';
+import { StudySetService } from './services/study-set.service';
```
[the remark this answers is not in the document]
Comment (ashkjones) [reply src/app/app.component.ts]: Done

File: src/app/edit-create-study-set/edit-create-study-set.component.ts [hunk 3]
```diff
@@ -12,6 +12,9 @@ import { MatIconModule } from '@angular/material/icon';
 import { FlashcardComponent } from '../flashcard/flashcard.component';
 import { FlashcardEditorComponent } from '../flashcard-editor/flashcard-editor.component';
 import { SequenceEditorComponent } from '../sequence-editor/sequence-editor.component';
+import { StudySetService } from '../services/study-set.service';
+import { StudySetData } from '../data-models/studyset-model';
+import { FlashcardData } from '../data-models/flashcard-model';
```
[the remark this answers is not in the document]
Comment (ashkjones) [reply src/app/edit-create-study-set/edit-create-study-set.component.ts]: I think I got them all.

File: src/app/app.component.ts [hunk 4]
```diff
@@ -5,13 +5,19 @@ import { MatSlideToggleModule } from '@angular/material/slide-toggle';
 import { FormsModule } from '@angular/forms';
 import { NavbarComponent } from './navbar/navbar.component';
 import { SignInComponent } from './sign-in/sign-in.component';
+import { EditCreateStudySetComponent } from './edit-create-study-set/edit-create-study-set.component';
```
[the remark this answers is not in the document]
Comment (ashkjones) [reply src/app/app.component.ts]: Sorry, I forgot to delete it after testing. I deleted it now.

**Comment by ashkjones** [comment]
I give my reviewer permission to make edits to my code directly if that is faster for them, instead of asking me to make the changes and then reviewing it again.

**Review by B-Yamawaki — APPROVED** [review]
Woo hoo. Very nice. Very good

**Pull request merged**